### PR TITLE
HTTP Basic auth for `connect`

### PR DESCRIPTION
**Pull request by mcpherrinm**
This uses credentials, if present, from the URL.

This code is manually tested, but unfortunately there's not really a test suite setup for connecting to things.  I think that's a much bigger project than just adding this, so I'm punting on tests til later.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/stretchr/testify v1.2.2 // indirect
 	golang.org/x/crypto v0.0.0-20181015023909-0c41d7ab0a0e
-	golang.org/x/sys v0.0.0-20181023152157-44b849a8bc13 // indirect
 	gopkg.in/alecthomas/kingpin.v2 v2.2.6
 	gopkg.in/asn1-ber.v1 v1.0.0-20170511165959-379148ca0225
 	gopkg.in/yaml.v2 v2.2.1 // indirect
 )
+
+go 1.13
```
Comment (mbyczkowski) [inline go.mod]: Looks like change is the same as #198, but the commit SHA is different. Maybe it needs another rebase?

File: starttls/dialer.go [hunk 2]
```diff
@@ -51,3 +55,19 @@ func dialWithDialer(dialer Dialer, timeout time.Duration, network, addr string,
...
 	return conn, nil
 }
+
+func wrapDialerWithProxy(dialer Dialer, connectProxy *url.URL, tlsConfig *tls.Config) (Dialer, error) {
+	dialerOpt := http_dialer.WithDialer(dialer.(*net.Dialer))
+	tlsOpt := http_dialer.WithTls(tlsConfig)
+	if connectProxy.User != nil {
+		password, ok := connectProxy.User.Password()
+		if !ok {
+			return nil, fmt.Errorf("proxy username without password not currently supported")
```
Comment (mbyczkowski) [inline starttls/dialer.go]: is anything special needed for proxying using a username without password? or does anyone even use that?

Comment (mcpherrinm) [reply starttls/dialer.go]: It's potentially possible (you could specify http://username@proxy/) but the upstream go-http-dialer package doesn't appear to support it (it always takes a username and password).

File: go.mod (hunk 1 above)
Comment (mcpherrinm) [reply go.mod]: yeah, will rebase or drop this commit

Comment (mcpherrinm) [reply go.mod]: rebased and it went poof

**Pull request merged**